src/StackExchange.Redis/ConfigurationOptions.cs

```diff
@@ -769,19 +739,153 @@ private void DoParse(string configuration, bool ignoreUnknown)
+                    ConfigurationChannel = value;
+                    break;
+                case OptionKeys.KeepAlive:
+                    KeepAlive = OptionKeys.ParseInt32(key, value);
+                    break;
+                case OptionKeys.ConnectTimeout:
+                    ConnectTimeout = OptionKeys.ParseInt32(key, value);
+                    break;
+                case OptionKeys.ConnectRetry:
+                    ConnectRetry = OptionKeys.ParseInt32(key, value);
+                    break;
+                case OptionKeys.ConfigCheckSeconds:
+                    ConfigCheckSeconds = OptionKeys.ParseInt32(key, value);
+                    break;
+                case OptionKeys.Version:
+                    DefaultVersion = OptionKeys.ParseVersion(key, value);
+                    break;
+                case OptionKeys.User:
+                    User = value;
+                    break;
+                case OptionKeys.Password:
+                    Password = value;
+                    break;
+                case OptionKeys.TieBreaker:
+                    TieBreaker = value;
+                    break;
+                case OptionKeys.Ssl:
+                    Ssl = OptionKeys.ParseBoolean(key, value);
+                    break;
+                case OptionKeys.SslHost:
+                    SslHost = value;
+                    break;
+                case OptionKeys.HighPrioritySocketThreads:
+                    HighPrioritySocketThreads = OptionKeys.ParseBoolean(key, value);
+                    break;
+                case OptionKeys.WriteBuffer:
+#pragma warning disable CS0618 // Type or member is obsolete
+                    WriteBuffer = OptionKeys.ParseInt32(key, value);
+#pragma warning restore CS0618 // Type or member is obsolete
+                    break;
+                case OptionKeys.Proxy:
+                    Proxy = OptionKeys.ParseProxy(key, value);
+                    break;
+                case OptionKeys.ResponseTimeout:
+#pragma warning disable CS0618 // Type or member is obsolete
+                    ResponseTimeout = OptionKeys.ParseInt32(key, value, minValue: 1);
+#pragma warning restore CS0618 // Type or member is obsolete
+                    break;
+                case OptionKeys.DefaultDatabase:
+                    DefaultDatabase = OptionKeys.ParseInt32(key, value);
+                    break;
+                case OptionKeys.PreserveAsyncOrder:
+                    break;
+                case OptionKeys.SslProtocols:
+                    SslProtocols = OptionKeys.ParseSslProtocols(key, value);
+                    break;
+                default:
+                    return false;
             }
+
+            return true;
+        }
+
+        private static Dictionary<string, HashSet<string>> ParseQueryString(string qs)
+        {
+            var dict = new Dictionary<string, HashSet<string>>();
+            foreach (var p in qs.Trim().Split(StringSplits.Ampersand))
```

Comment:
value, and probably the key, need unescaping, presumably; consider %-encoded parameters

Reply:
done